crates/uv/src/commands/tool/install.rs

```diff
@@ -103,14 +103,19 @@ pub(crate) async fn install(
 
     let requirements = [Ok(from.clone())]
         .into_iter()
-        .chain(with.iter().map(|name| Requirement::from_str(name)))
-        .collect::<Result<Vec<Requirement<VerbatimParsedUrl>>, _>>()?;
+        .chain(
+            with.iter()
+                .map(|name| pep508_rs::Requirement::from_str(name)),
+        )
+        .collect::<Result<Vec<pep508_rs::Requirement<VerbatimParsedUrl>>, _>>()?;
 
-    // TODO(zanieb): Duplicative with the above parsing but needed for `update_environment`
-    let requirements_sources = [RequirementsSource::from_package(from.to_string())]
-        .into_iter()
-        .chain(with.into_iter().map(RequirementsSource::from_package))
-        .collect::<Vec<_>>();
+    let spec = RequirementsSpecification::from_requirements(
+        requirements
+            .iter()
+            .cloned()
+            .map(pypi_types::Requirement::from)
+            .collect(),
+    );
```

Comment:
Ah thank you! I need a guide on our various requirement types haha

Reply:
I'm gonna audit the usages in these commands, all good.

Reply:
(But, as a separate PR.)